src/EFCore.PG/Extensions/NpgsqlNetworkExtensions.cs

```diff
@@ -0,0 +1,1006 @@
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static IPAddress Subtract([CanBeNull] this DbFunctions _, IPAddress inet, int value) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Subtracts the <paramref name="value"/> from the (IPAddress Address, int Subnet).
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="cidr">The inet.</param>
+        /// <param name="value">The value to subtract.</param>
+        /// <returns>
+        /// The (IPAddress Address, int Subnet) augmented by the <paramref name="value"/>.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static (IPAddress Address, int Subnet) Subtract([CanBeNull] this DbFunctions _, (IPAddress Address, int Subnet) cidr, int value) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Subtracts one <see cref="IPAddress"/> from another <see cref="IPAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="inet">The inet from which to subtract.</param>
+        /// <param name="other">The inet to subtract.</param>
+        /// <returns>
+        /// The <see cref="IPAddress"/> augmented by the <paramref name="other"/>.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static IPAddress Subtract([CanBeNull] this DbFunctions _, IPAddress inet, IPAddress other) => throw new ClientEvaluationNotSupportedException();
+
+        /// <summary>
+        /// Subtracts one (IPAddress Address, int Subnet) from another (IPAddress Address, int Subnet).
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="cidr">The cidr from which to subtract.</param>
+        /// <param name="other">The cidr to subtract.</param>
+        /// <returns>
+        /// The (IPAddress Address, int Subnet) augmented by the <paramref name="other"/>.
+        /// </returns>
+        /// <exception cref="ClientEvaluationNotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static (IPAddress Address, int Subnet) Subtract([CanBeNull] this DbFunctions _, (IPAddress Address, int Subnet) cidr, (IPAddress Address, int Subnet) other) => throw new ClientEvaluationNotSupportedException();
+
+        #endregion
+
+        #region Functions
```

Comment:
I realize that the `cidr` tuple has `Subnet`, but I think the methods below should be named as close as possible to ther PostgreSQL equivalents (aside from obvious casing differences with C#), so I'd retain `MaskLength()` rather than `SubnetLength()` etc.